WeakAuras/RegionTypes/DynamicGroup.lua

```diff
@@ -940,6 +958,50 @@ local function SafeGetPos(region, func)
   end
 end
 
+local function isDifferent(regionData, cache, events)
+  local id = regionData.id
+  local cloneId = regionData.cloneId or ""
+  local state = regionData.region.state
+  if not events then
+    return false
+  elseif events.changed then
+    return true -- escape hatch, not super recommended
+  else
+    local isDifferent = false
+    if not cache[id] then
+      isDifferent = true
+      local cachedState = {}
+      cache[id] = {[cloneId] = cachedState}
+      for event in pairs(events) do
+        cachedState[event] = state[event]
+      end
+    elseif not cache[id][cloneId] then
+      isDifferent = true
+      local cachedState = {}
+      cache[id][cloneId] = cachedState
+      for event in pairs(events) do
+        cachedState[event] = state[event]
+      end
+    else
+      local cachedState = cache[regionData.id][regionData.cloneId]
```

Comment:
```suggestion
      local cachedState = cache[id][cloneId]
```
prevent error if regionData.cloneId is nil